File: server/internal/model/invitation.go
package model

import (
	"time"

	"github.com/google/uuid"
)

type WorkspaceInvitation struct {
	ID         uuid.UUID  `json:"id"`
	InviterID  uuid.UUID  `json:"inviter_id"`
	Email      *string    `json:"email,omitempty"`
	Token      string     `json:"token"`
	ExpiresAt  time.Time  `json:"expires_at"`
	AcceptedAt *time.Time `json:"accepted_at,omitempty"`
	AcceptedBy *uuid.UUID `json:"accepted_by,omitempty"`
	MaxUses    int        `json:"max_uses"`
	UseCount   int        `json:"use_count"`
	CreatedAt  time.Time  `json:"created_at"`
	InviteURL  string     `json:"invite_url,omitempty"`
}

type CreateInvitationRequest struct {
	Email   *string `json:"email" validate:"omitempty,email,max=255"`
	MaxUses int     `json:"max_uses" validate:"min=1,max=1000"`
	TTLDays int     `json:"ttl_days" validate:"min=1,max=30"`
}

type AcceptInvitationRequest struct {
	Token string `json:"token" validate:"required"`
}

File: desktop/src/types/invitation.ts
export interface WorkspaceInvitation {
  id: string;
  inviter_id: string;
  email?: string;
  token: string;
  expires_at: string;
  accepted_at?: string;
  accepted_by?: string;
  max_uses: number;
  use_count: number;
  created_at: string;
  invite_url: string;
}

export interface CreateInvitationRequest {
  email?: string;
  max_uses: number;
  ttl_days: number;
}
